homeassistant/components/ecobee.py

```diff
@@ -15,12 +16,19 @@
 
 DOMAIN = "ecobee"
 NETWORK = None
-HOLD_TEMP = 'hold_temp'
+CONF_HOLD_TEMP = 'hold_temp'
 
 REQUIREMENTS = [
     'https://github.com/nkgilley/python-ecobee-api/archive/'
     '4856a704670c53afe1882178a89c209b5f98533d.zip#python-ecobee==0.0.6']
 
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_API_KEY): str,
```

Comment:
According to the docs/code should the API key be marked as `vol.Required`. How about using `cv.string` here too.